test/framework/pod.go

```diff
@@ -60,3 +68,59 @@ func (f *Framework) GetPodRestartCount(ns, podName string) (map[string]int32, er
 
 	return restarts, nil
 }
+
+// ExecOptions passed to ExecWithOptions
+type ExecOptions struct {
+	Command       []string
+	Namespace     string
+	PodName       string
+	ContainerName string
+	Stdin         io.Reader
+	CaptureStdout bool
+	CaptureStderr bool
+	// If false, whitespace in std{err,out} will be removed.
+	PreserveWhitespace bool
+}
+
+// ExecWithOptions executes a command in the specified container,
```

Comment:
Is this borrowed from Kubernetes? :) If yes let's mention that? Anything against importing it instead?

Reply:
importing k/k is a nightmare, I'll give a reference where this is from yes